### PR TITLE
Reduce allocations in `g_log` bindings

### LINKED ISSUE #373: Reduce allocations in `g_log` bindings
https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/906/diffs?commit_id=74217a02d5fea72ae8c0934cb4add4229a40334a#d9672eb1cfd376b4abb6e7743ff284a5605bb55c_183_204

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #373.

### CODE REVIEW

File: glib/src/log.rs [hunk 1]
```diff
@@ -423,25 +406,6 @@ macro_rules! g_log {
...
-        use $crate::translate::{IntoGlib, ToGlibPtr};
-        use $crate::LogLevel;
-
-        fn check_log_args(_log_level: LogLevel, _format: &str) {}
-
-        // the next line is used to enforce the type for the macro checker...
-        let log_domain: Option<&str> = $log_domain.into();
-        check_log_args($log_level, $format);
-        // to prevent the glib formatter to look for arguments which don't exist
-        let f = $format.replace("%", "%%");
```
Comment (sdroege) [inline glib/src/log.rs:436]: Same here

File: glib/src/log.rs [hunk 2]
```diff
@@ -387,24 +387,7 @@ macro_rules! g_log {
...
     }};
-    ($log_level:expr, $format:literal $(,)?) => {{
-        use $crate::translate::{IntoGlib, ToGlibPtr};
-        use $crate::LogLevel;
-
-        fn check_log_args(_log_level: LogLevel, _format: &str) {}
-
-        check_log_args($log_level, $format);
-        // to prevent the glib formatter to look for arguments which don't exist
-        let f = $format.replace("%", "%%");
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply glib/src/log.rs:398]: You mean one more, no?

**Comment by GuillaumeGomez** [comment]
I added the allocation reduction directly into this PR here so that it makes sense as a whole.

File: glib/src/log.rs [hunk 3]
```diff
@@ -370,77 +422,14 @@ pub fn log_default_handler(log_domain: Option<&str>, log_level: LogLevel, messag
...
-        unsafe {
-            $crate::ffi::g_log(
-                log_domain.to_glib_none().0,
-                $log_level.into_glib(),
-                f.to_glib_none().0,
-            );
-        }
```
Comment (sdroege) [inline glib/src/log.rs:443]: Same many lines below for `g_print`. Search for `%%` in this file :)

Looks good to me otherwise

**Comment by GuillaumeGomez** [comment]
Added the change into `g_print` macros as well.

**Review by sdroege — APPROVED** [review]

**Pull request merged**